waiter/integration/waiter/grpc_test.clj

```diff
@@ -45,65 +53,221 @@
   [length]
   (apply str (take length (repeatedly #(char (+ (rand 26) 65))))))
 
-(deftest ^:parallel ^:integration-fast test-basic-grpc-server
-  (testing-using-waiter-url
-    (GrpcClient/setLogFunction (reify Function
-                                 (apply [_ message]
-                                   (log/info message))))
-    (let [[host _] (str/split waiter-url #":")
-          h2c-port (Integer/parseInt (retrieve-h2c-port waiter-url))
-          request-headers (basic-grpc-service-parameters)
-          {:keys [cookies headers] :as response} (make-request waiter-url "/waiter-ping" :headers request-headers)
-          cookie-header (str/join "; " (map #(str (:name %) "=" (:value %)) cookies))
-          service-id (get headers "x-waiter-service-id")]
+(defn ping-courier-service
+  [waiter-url request-headers]
+  (make-request waiter-url "/waiter-ping" :headers request-headers))
+
+(defn start-courier-instance
+  [waiter-url]
+  (let [[host _] (str/split waiter-url #":")
+        h2c-port (Integer/parseInt (retrieve-h2c-port waiter-url))
+        request-headers (basic-grpc-service-parameters)
+        {:keys [cookies headers] :as response} (ping-courier-service waiter-url request-headers)
+        cookie-header (str/join "; " (map #(str (:name %) "=" (:value %)) cookies))
+        service-id (get headers "x-waiter-service-id")
+        request-headers (assoc request-headers
+                          "cookie" cookie-header
+                          "x-waiter-timeout" "60000")]
+    (assert-response-status response 200)
+    (is service-id)
+    (log/info "ping cid:" (get headers "x-cid"))
+    (log/info "service-id:" service-id)
+    (let [{:keys [ping-response service-state]} (some-> response :body try-parse-json walk/keywordize-keys)]
+      (is (= "received-response" (:result ping-response)) (str ping-response))
+      (is (str/starts-with? (str (some-> ping-response :headers :server)) "courier-health-check"))
+      (assert-response-status ping-response 200)
+      (is (or (= {:exists? true :healthy? true :service-id service-id :status "Running"} service-state)
+              (= {:exists? true :healthy? false :service-id service-id :status "Starting"} service-state))
```

Comment:
Is it possible to receive a valid health check response from a service in the `Starting` state?